### ISSUE TITLE
Data is deleted when opening the page in another tab

### ISSUE THREAD

**Issue by azat-io** [nanostores/persistent#14]
You can try to check the problem here:
https://si5dw.csb.app/

Source code:
https://codesandbox.io/s/nanostores-i18n-si5dw

Steps to reproduce:
1. Set state with buttons
2. Open page in another tab

**Comment by ai** [comment]
Sorry for long delay. Fixed d753470.

**Comment by azat-io** [comment]
@ai Thanks a lot!

**Comment by ai** [comment]
I found another regression in demo

**Comment by admir4l** [comment]
Hi
I am using this library to persist my store, but the state is cleared each time I open a new tab.

According to the code, the code does not check for the equality of key values of fired events, and that is the major problem!

My next step will be to rewrite the whole file. If that is helpful, I can post it here

**Comment by ai** [comment]
@admir4l Can you send PR? I was not able to localize the issue on local machine to debug `node_modules`. Your help will be very useful!

**Comment by admir4l** [comment]
@ai  
I appreciate your great library. You can find a PR for the issue here #20 .

I'm also restructuring this to use the localstorage methods instead of direct access to the storage object, but it will break compatibility with previous versions!
[https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage)

If you think that is useful, I can create another PR for that once it is ready

**Comment by ai** [comment]
> use the localstorage methods instead of direct access to the storage object

What is benefits of using methods?

**Comment by admir4l** [comment]
I believe they are, basically, exactly the same.
However, the MDN says:

> The localStorage read-only property of the window interface allows you to access a Storage object for the Document's origin

You can read more discussion about this [here](https://stackoverflow.com/questions/13092715/localstorage-use-getitem-setitem-functions-or-access-object-directly).

**Comment by ai** [comment]
Let’s keep current logic if there is no rational reasons
